sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1608,6 +1608,47 @@ class AdaptiveQueryExecSuite
     }
   }
 
+  test("SPARK-49460: NPE error in EmptyRelationExec.cleanupResources") {
```

Comment:
nit: `NPE` already refers to `NullPointerException`, so the `error` in the naming is somewhat redundant.